packages/orm/src/createOrUpdatePartial.ts

```diff
@@ -122,37 +122,41 @@ export async function createOrUpdatePartial<T extends Entity>(
+        toArray(value).map(async (value: any) => {
+          if (!value || isEntity(value)) {
+            return value;
+          } else if (isKey(value)) {
+            return await em.load(field.otherMetadata().cstr, value);
+          } else {
+            // Look for `delete: true/false` and `remove: true/false` markers
+            const deleteMarker: any = allowDelete && value["delete"];
+            const removeMarker: any = allowRemove && value["remove"];
+            const opMarker: any = allowOp && value["op"];
+            // If this is the incremental marker, just leave it in as-is so that setOpts can see it
+            if (opMarker === "incremental") {
               return value;
-            } else if (isKey(value)) {
-              return await em.load(field.otherMetadata().cstr, value);
-            } else {
-              // Look for `delete: true/false` and `remove: true/false` markers
-              const deleteMarker: any = allowDelete && value["delete"];
-              const removeMarker: any = allowRemove && value["remove"];
-              const opMarker: any = allowOp && value["op"];
-              // If this is the incremental marker, just leave it in as-is so that setOpts can see it
-              if (opMarker === "incremental") {
-                return value;
-              }
-              // Remove the markers, regardless of true/false, before recursing into createOrUpdatePartial to avoid unknown fields
-              if (deleteMarker !== undefined) delete value.delete;
-              if (removeMarker !== undefined) delete value.remove;
-              if (opMarker !== undefined) delete value.op;
-              const entity = await createOrUpdatePartial(em, field.otherMetadata().cstr, value as any);
-              // Put the markers back for setOpts to find
-              if (deleteMarker === true) entity.delete = true;
-              if (removeMarker === true) entity.remove = true;
-              if (opMarker) entity.op = opMarker;
-              return entity;
             }
-          });
-      return [name, await Promise.all(entities)];
+            // Remove the markers, regardless of true/false, before recursing into createOrUpdatePartial to avoid unknown fields
+            if (deleteMarker !== undefined) delete value.delete;
+            if (removeMarker !== undefined) delete value.remove;
+            if (opMarker !== undefined) delete value.op;
+            const entity = await createOrUpdatePartial(em, field.otherMetadata().cstr, value as any);
+            // Put the markers back for setOpts to find
+            if (deleteMarker === true) entity.delete = true;
+            if (removeMarker === true) entity.remove = true;
+            if (opMarker) entity.op = opMarker;
+            return entity;
+          }
+        }),
+      );
+      return [
+        name,
+        // Unwrap & keep null for o2o fields
+        field.kind === "o2o" ? (value === null ? null : entities[0]) : entities,
```

Comment:
This is new, to unwrap o2os out of the collection, since I wanted to mostly keep the ^ generic handling.

...that said, all of the ^ generic handling is around delete/remove/op markers, which o2os don't support/don't need, so maybe I should have just had totally separate `else` block anyway...